app/api/aggregate_outbreaks.py

```diff
@@ -330,3 +330,43 @@ def add_non_open_rows_for_each_open_row(df):
     processed_df = df.groupby(group_cols, as_index=False).apply(
         lambda x: add_non_open_row_if_necessary(x, df, col_map))
     return processed_df
+
+
+# accepts a group of facilities as df, building up a running total of cumulative values from past closed outbreaks
+# when new outbreaks open, add in the running total of cumulative values from past outbreaks
+def process_sum_outbreaks(df):
+    df.sort_values(by=['Date'], ignore_index=True, inplace=True)
+    col_map = utils.make_matching_column_name_map(df)
+    running_cumulative_values = None
+
+    def apply_sum(row):
+        index = row.name
+        nonlocal running_cumulative_values
+
+        # if our row is open and the above row is closed, look up at the past closed outbreak
+        if index-1 > 0 and row["Outbrk_Status"].upper() == "OPEN" \
+                and df.iloc[index-1]["Outbrk_Status"].upper() == "CLOSED":
```

Comment:
will it always say "CLOSED" or can it be blank sometimes which also means closed?

Reply:
Good to check that. It will always say CLOSED at this point because we did a step in `post_processing`(`close_unknown_outbreaks=True`) where everything with a blank outbreak status is set to closed, which Artis tells me is the right thing to do here.